pyiges/check_imports.py

```diff
@@ -0,0 +1,20 @@
+
+try:
+    import pyvista
+    import geomdl
+    from pyvista._vtk import vtkAppendPolyData
+    _IS_FULL_MODULE = True
+except (ModuleNotFoundError, ImportError) as exc:
+    pyvista = None
+    geomdl = None
+    vtkAppendPolyData = None
+    _IS_FULL_MODULE = False
+    _PROBLEM_MSG = 'Import from \'{}\' failed, to support this feature '\
+        'please install pyiges[full]'.format(exc.name)
+
+def assert_full_module_variant(inner_func):
+    def safe_func(*a, **kw):
+        if not _IS_FULL_MODULE:
+            raise Exception(_PROBLEM_MSG)
+        return inner_func(*a, **kw)
+    return safe_func
```

Comment:
@banesullivan This new file is the critical point and serves as an "import interface". If the modules are not imported, the decorator `assert_full_module_variant` will raise an appropriate exception. Note that this approach is quite close to Your suggestion made here https://github.com/pyvista/pyiges/issues/28#issuecomment-1497804254 

The drawback of this approach, is that one can "forget" to decorate a function, but the approach is pragmatic IMO